### PR TITLE
Feat/completions delete s3 att

### PR DESCRIPTION
**Pull request by dannil76**
## Explain the changes you’ve made
Added helper function for deleting attachments after completions is submitted to Viva

## Explain why these changes are made
The file has no relevnace to be stored in S3 after it's received by Viva. It also increase the personal privacy.

## Explain your solution
Implemented the deleteAttachments function. It will map over all attachments and return an list of promises to be resloved at once

## How to test
1. Checkout this branch
2. run `sls deploy` within service folder for `viva-ms`
3. Complete an completions application and verify that all releated files is deleted from users folder in the S3 bucket 'attachments'

## Tested environments

- [X] Your personal AWS environment.
- [] Your local Serverless environment.
---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-234a5vx submitCompletions-lambda ska kunna radera filer efter de är mottagna av Vada](https://app.clickup.com/t/234a5vx)

File: services/viva-ms/src/lambdas/submitCompletion.js [hunk 1]
```diff
@@ -131,9 +130,15 @@ export async function main(event, context) {
...
   }
 
+  await deleteAttachments(attachmentList);
+
   return true;
 }
 
+function deleteAttachments(attachmentList) {
+  return Promise.all(attachmentList.map(({ id }) => S3.deleteFile(process.env.BUCKET_NAME, id)));
+}
```
Comment (kajm) [inline services/viva-ms/src/lambdas/submitCompletion.js:140]: Perhaps:
```js
function deleteAttachment(attachment) {
  return S3.deleteFile(process.env.BUCKET_NAME, attachment.id)
}
```

and used like:
```js
const deleteAttachmentPromises = attachmentList.map(deleteAttachment);
await Promise.all(deleteAttachmentPromises);
```

Comment (kajm) [reply services/viva-ms/src/lambdas/submitCompletion.js:140]: Was wondering if we shouldn't use `Promise.all` in this case. If any of the promises rejects then every other promise will reject as well. Perhaps we should use `allSettled` instead and log if any file on s3 fails to be deleted.

File: services/viva/microservice/test/lambdas/submitCompletion.test.ts [hunk 2]
```diff
@@ -62,12 +62,32 @@ function createDependencies(
     postCompletion: () => Promise.resolve({ status: 'OK' }),
     updateCase: () => Promise.resolve(),
     getAttachments: () => Promise.resolve([]),
+    deleteAttachments: () => Promise.resolve(undefined),
     ...partialDependencies,
   };
 }
 
 it('successfully submits completions', async () => {
```
Comment (dannil76) [inline services/viva/microservice/test/lambdas/submitCompletion.test.ts:70]: maybe add '...and deletes attachments after success'?

Comment (kajm) [reply services/viva/microservice/test/lambdas/submitCompletion.test.ts:70]: Will not change the description.

File: services/viva/microservice/src/lambdas/submitCompletion.ts [hunk 3]
```diff
@@ -85,9 +88,16 @@ async function updateCase(params: UpdateCaseParameters): Promise<void> {
...
+  const keys = attachments.map(({ id }) => ({ Key: id }));
+
+  return S3.deleteFiles(process.env.BUCKET_NAME as string, keys);
+}
+
 export async function submitCompletion(input: LambdaRequest, dependencies: Dependencies) {
   const { caseKeys } = input.detail;
-  const { getCase, readParams, postCompletion, updateCase, getAttachments } = dependencies;
+  const { getCase, readParams, postCompletion, updateCase, getAttachments, deleteAttachments } =
+    dependencies;
```
[the remark this answers is not in the document]
Comment (kajm) [reply services/viva/microservice/src/lambdas/submitCompletion.ts:100]: fixec

**Review by dannil76 — COMMENTED** [review]
Cool

**Review by dannil76 — COMMENTED** [review]
Offcoruse

**Pull request merged**